resources/charts/kubelet/templates/kubelet-ds.yaml

```diff
@@ -0,0 +1,137 @@
+          --cni-conf-dir=/etc/kubernetes/cni/net.d \
+          --config=/etc/kubernetes/kubelet.config \
+          --kubeconfig=/etc/kubernetes/kubeconfig \
+          --lock-file=/var/run/lock/kubelet.lock \
+          --network-plugin=cni \
+          --pod-manifest-path=/etc/kubernetes/manifests \
+          --read-only-port=0 \
+          --volume-plugin-dir=/var/lib/kubelet/volumeplugins \
+          --node-labels=$(grep NODE_LABELS /etc/kubernetes/kubelet.env | cut -d'"' -f2) \
+          --register-with-taints=$(grep NODE_TAINTS /etc/kubernetes/kubelet.env | cut -d'"' -f2)
+          " | tee /tmp/start-kubelet.sh &&
+          chmod +x /tmp/start-kubelet.sh &&
+          /tmp/start-kubelet.sh
+        env:
+        - name: HOST_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - mountPath: /var/lib/cni
+          name: coreos-var-lib-cni
+          readOnly: false
+        - mountPath: /var/lib/calico
+          name: coreos-var-lib-calico
+          readOnly: true
+        - mountPath: /opt/cni/bin
+          name: coreos-opt-cni-bin
+          readOnly: true
+        # TODO check if this is needed
+        - name: dev
+          mountPath: /dev
+        - name: run
+          mountPath: /run
+        - name: sys
+          mountPath: /sys
+          readOnly: true
+        - name: etc-kubernetes
+          mountPath: /etc/kubernetes
+          readOnly: true
+        - name: var-lib-docker
+          mountPath: /var/lib/docker
+        - name: var-lib-kubelet
+          mountPath: /var/lib/kubelet
+          mountPropagation: Bidirectional
+          # Logs from the kubelet pods
+        - name: logs
+          mountPath: /var/log/pods
+          # This is mounted from host to make sure that the kubelet showcases OS as Flatcar and not
+          # Debian from the hyperkube image.
+        - name: os-release
+          mountPath: /etc/os-release
+          readOnly: true
+      hostNetwork: true
+      hostPID: true
+      # Tolerate all the taints. This ensures that the pod runs on all the nodes.
+      # https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+      tolerations:
+      - operator: "Exists"
+      volumes:
+      - name: coreos-var-lib-cni
+        hostPath:
+          path: /var/lib/cni
+      - name: coreos-var-lib-calico
+        hostPath:
+          path: /var/lib/calico
+      - name: coreos-opt-cni-bin
+        hostPath:
+          path: /opt/cni/bin
+      - name: dev
+        hostPath:
+          path: /dev
+      - name: run
+        hostPath:
+          path: /run
+      - name: sys
+        hostPath:
+          path: /sys
+      - name: etc-kubernetes
+        hostPath:
+          path: /etc/kubernetes
+      - name: var-lib-docker
+        hostPath:
+          path: /var/lib/docker
+      - name: var-lib-kubelet
+        hostPath:
+          path: /var/lib/kubelet
```

Comment:
I'm wondering if it should have the `mountPropagation: Bidirectional`, as this is needed when running `kubelet` as Docker container.

Reply:
What do you mean? In self hosted kubelet aren't we running kubelet inside a docker container?

Reply:
Yes, but kubelet also performs some cleanups on pods, so it needs to see the mounts done by the Docker daemon. Hence the need of the propagation. Are you sure other pods are created and removed without any errors printed by `kubelet`?